src/store.ts

```diff
@@ -140,28 +181,37 @@ export class Store<T> {
     }
   }
 
-  private async internalDispatch(reducer: Reducer<T>, ...params: any[]) {
-    if (!this.actions.has(reducer)) {
-      throw new Error(`Tried to dispatch an unregistered action${reducer ? " " + reducer.name : ""}`);
+  private async internalDispatch(actions: DispatchAction<T>[]) {
+    const unregisteredAction = actions.find((a) => !this.actions.has(a.reducer));
+    if (unregisteredAction) {
+      throw new Error(`Tried to dispatch an unregistered action ${unregisteredAction.reducer && unregisteredAction.reducer.name}`);

```

Comment:
since we have now the same error thrown before, this part never gets really touched by tests.

Reply:
True. I added an extra describe block, testing if the internalDispatch throws the error correctly, bypassing the visibility with an any-cast as that seems to be the primary way for library users to bypass the first check.